x/merkledb/db.go

```diff
@@ -483,19 +486,11 @@ func (db *merkleDB) PrefetchPath(key []byte) error {
 
 func (db *merkleDB) prefetchPath(keyBytes []byte) error {
 	return visitPathToKey(db, ToKey(keyBytes), func(n *node) error {
-		if !n.hasValue() {
-			// this value is already in the cache, so skip writing
```

Comment:
the lru cache does lock on the Get call, so there is no benefit of checking presence first.

Reply:
The benefit is that the Get call only uses an RLock, so this can be much more concurrent friendly.

Reply:
https://github.com/ava-labs/avalanchego/blob/master/cache/lru_sized_cache.go#L41